sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/BinaryComparisonSimplificationSuite.scala

```diff
@@ -119,9 +119,7 @@ class BinaryComparisonSimplificationSuite extends PlanTest with PredicateHelper
 
     // GetStructField with different names are semantically equal; thus, `EqualTo(fieldA1, fieldA2)`
     // will be optimized to `TrueLiteral` by `SimplifyBinaryComparison`.
-    val originalQuery = nonNullableRelation
-        .where(EqualTo(fieldA1, fieldA2))
-        .analyze
+    val originalQuery = nonNullableRelation.where(EqualTo(fieldA1, fieldA2))
```

Comment:
Here we construct the resolved plan directly, because we need to reply on case insensitive and `.analyze` use case sensitive analyzer.

Reply:
I go to look at this test and its original PR. `fieldA1` and `fieldA2` are not different in letter case, but the `name` in `GetStructField`. That is why above comment is `GetStructField with different names are semantically equal`.